pkg/operator/staticpod/certsyncpod/certsync_cmd.go

```diff
@@ -82,7 +82,7 @@ func (o *CertSyncControllerOptions) Run() error {
 
 	kubeInformers := informers.NewSharedInformerFactoryWithOptions(o.kubeClient, 10*time.Minute, informers.WithNamespace(o.Namespace))
 
-	eventRecorder := events.NewKubeRecorder(o.kubeClient.CoreV1().Events(o.Namespace), "cert-syncer",
+	eventRecorder := events.NewKubeRecorder(o.kubeClient.CoreV1().Events(o.Namespace), events.RecommendedClusterSingletonCorrelatorOptions(), "cert-syncer",
```

Comment:
award for the longest identifier! 🏆

Reply:
goes to @deads2k for suggesting it :)

Reply:
it has some Java smell, am not surprised :D